src/components/WrapEtherBtn.tsx

```diff
@@ -0,0 +1,269 @@
+import React, { useMemo } from 'react'
+import styled from 'styled-components'
+import { DEFAULT_MODAL_OPTIONS, ModalBodyWrapper } from 'components/Modal'
+import Modali, { useModali } from 'modali'
+import { InputBox } from 'components/InputBox'
+import { useForm } from 'react-hook-form'
+import { DEFAULT_PRECISION, formatAmountFull } from '@gnosis.pm/dex-js'
+import BN from 'bn.js'
+import { validatePositiveConstructor, validInputPattern } from 'utils'
+import { TooltipWrapper } from 'components/Tooltip'
+import useSafeState from 'hooks/useSafeState'
+
+export const INPUT_ID_AMOUNT = 'wrapAmount'
+
+const ModalWrapper = styled(ModalBodyWrapper)`
+  > form {
+    > div {
+      margin: 3rem 1.5rem;
+
+      ${InputBox} {
+        position: relative;
+        display: flex;
+        flex-flow: row wrap;
+
+        i {
+          position: absolute;
+          right: 1rem;
+          top: 0;
+          bottom: 0;
+          margin: 0;
+          color: #476481;
+          letter-spacing: -0.05rem;
+          text-align: right;
+          font-family: var(--font-default);
+          font-weight: var(--font-weight-bold);
+          font-size: 1.2rem;
+          font-style: normal;
+          display: flex;
+          justify-content: center;
+          align-items: center;
+        }
+      }
+    }
+
+    .error {
+      color: red;
+    }
+
+    b {
+      font-size: 1.3rem;
+      color: #2f3e4e;
+      margin: 0 1.6rem 0 0;
+      margin-bottom: 0.5rem;
+      padding-left: -0.5;
+      display: block;
+    }
+
+    a {
+      color: rgb(33, 141, 255);
+    }
+
+    p a {
+      font-size: 1.2rem;
+      margin-left: 0.2rem;
+    }
+
+    .more-info {
+      font-size: 1.3rem;
+      padding-left: 1rem;
+    }
+  }
+`
+
+interface WrapUnwrapEtherBtnProps {
+  wrap: boolean
+  label?: string
+  className?: string
+}
+
+export type WrapEtherBtnProps = Omit<WrapUnwrapEtherBtnProps, 'wrap'>
+
+interface WrapUnwrapInfo {
+  title: string
+  symbolSource: string
+  balance: BN
+  tooltipText: JSX.Element | string
```

Comment:
I think it's recommended to use `React.ReactNode` over `JSX.Element` (also below)